src/get-env-variables.js

```diff
@@ -0,0 +1,60 @@
+const exec = require('child_process').exec
+// import spawn from 'spawn-command-with-kill'
+import {assign, clone} from 'lodash'
+import managePath from 'manage-path'
+import {sync as findUpSync} from 'find-up'
+
+const NON_ERROR = 0
+
+export default getEnv
+
+function getEnv() {
+  return new Promise((resolve, reject) => {
+    getNpmEnvVariables()
+      .then(npmVars => {
+        const env = assign(clone(process.env), npmVars)
+        const alterPath = managePath(env)
+        const npmBin = findUpSync('node_modules/.bin')
+        if (npmBin) {
+          alterPath.unshift(npmBin)
+        }
+        resolve(env)
+      })
+      .catch(reject)
+  })
+}
+
+function getNpmEnvVariables() {
+  let child
+  return new Promise((resolve, reject) => {
+    let output = ''
+    child = exec('npm run env')
```

Comment:
Is this the only way to get the environment variables? Could you perhaps check npm's code to see whether they expose this via a node API? Would be really nice to not have to spawn anything...